src/runtime_src/core/tools/xbtracer/src/lib/capture.h

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "experimental/xrt_elf.h"
+#include "experimental/xrt_hw_context.h"
+#include "experimental/xrt_ip.h"
+#include "experimental/xrt_module.h"
+#include "experimental/xrt_system.h"
+#include "experimental/xrt_xclbin.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_graph.h"
+#include "xrt/xrt_kernel.h"
+
+
+#include "xrt_device_inst.h"
+
+namespace xrt::tools::xbtracer {
+
+class xrt_ftbl
+{
+  public:
+  xrt_device_ftbl     device;
+
+  static xrt_ftbl& get_instance()
+  {
+    static xrt_ftbl instance;
+    return instance;
+  }
```

Comment:
hmm, will this work as expected in a header file?  What if this header is included in multiple compilation units.?  Maybe the linker will handle it, but I would implement get_instance() in capture.cpp rather than in header.

Also, I don't think xrt_ftbl should be a class.  This is a singleton, a namespace would suffice with a method that returns what you need.

In header:
```
namespace xrt_ftbl {

xrt_device_ftbl
get_tbl();

}
```

in cpp:
```
namespace xrt_core::tools::xbtracer {
...

namespace xrt_ftbl {
xrt_device_ftbl* // or reference 
get_tbl()
{
   static xrt_device_ftbl device;
   return &device;
}
} // xrt_ftbl

....
} // xrt_core::tools::xbtracer
```

Reply:
I have made changes related to defining get_instance(). But has still kept xrt_ftbl as class. Please see if this would alright or suggest if I should indeed change it to namespace.